lib/Reconciler.lua

```diff
@@ -149,39 +151,52 @@ end
 		- `parent`: The Roblox object to contain the contained instances
 		- `key`: The Name to give the Roblox instance that gets created
 		- `context`: Used to pass Roact context values down the tree
+		- `parentElement`: Used for diagnostics. Refers to the parent Roact element.
 
 	The structure created by this method is important to the functionality of
 	the reconciliation methods; they depend on this structure being well-formed.
 ]]
-function Reconciler._reifyInternal(element, parent, key, context)
+function Reconciler._reifyInternal(element, parent, key, context, parentElement)
 	if isPrimitiveElement(element) then
 		-- Primitive elements are backed directly by Roblox Instances.
 
 		local rbx = Instance.new(element.component)
 
+		-- This name can be passed through multiple components.
+		-- What's important is the final Roblox Instance receives the name
+		-- It's solely for debugging purposes; Roact doesn't use it.
+		-- It must be set prior to children or property
+		if key then
+			rbx.Name = key
+		end
+
+		-- Only set these properties if tracing is enabled, since that's what
+		-- these are used for.
+		if GlobalConfig.getValue("logAllMutations") then
+			element._rbx = rbx
+			element._parent = parentElement
+			element._rbxParent = parent
+		end
```

Comment:
This doesn't actually work -- you can reuse elements to create multiple trees and breaks the immutability of elements in a pretty bad way.

We should be modifying the component *instances* or *instance handles*. They already have these kinds of values and are stateful and mutable.